centreon/src/Core/ServiceCategory/Infrastructure/Repository/DbReadServiceCategoryRepository.php

```diff
@@ -489,33 +489,182 @@ private function createServiceCategoryFromArray(array $result): ServiceCategory
     }
 
     /**
-     * @param SqlConcatenator $concatenator
+     * @param int[] $accessGroupIds
+     *
+     * @return array<SqlConcatenator>
+     */
+    private function findServiceCategoriesRequest(array $accessGroupIds = []): array
+    {
+        $hostAcls = '';
+        $hostGroupAcls = '';
+        $hostCategoryAcls = '';
+        if ([] !== $accessGroupIds) {
+            if (! $this->hasAccessToAllHosts($accessGroupIds)) {
+                $hostAcls = <<<'SQL'
+                    AND hsr.host_host_id IN (
+                        SELECT arhr.host_host_id
+                        FROM `:db`.acl_resources_host_relations arhr
+                        INNER JOIN `:db`.acl_resources res
+                            ON res.acl_res_id = arhr.acl_res_id
+                        INNER JOIN `:db`.acl_res_group_relations argr
+                            ON argr.acl_res_id = res.acl_res_id
+                        INNER JOIN `:db`.acl_groups ag
+                            ON ag.acl_group_id = argr.acl_group_id
+                        WHERE ag.acl_group_id IN (:access_group_ids)
+                    )
+                    SQL;
+            }
+
+            if (! $this->hasAccessToAllHostGroups($accessGroupIds)) {
+                $hostGroupAcls = <<<'SQL'
+                    AND hr.hostgroup_hg_id IN (
+                        SELECT arhgr.hg_hg_id
+                        FROM `:db`.acl_resources_hg_relations arhgr
+                        INNER JOIN `:db`.acl_resources res
+                            ON res.acl_res_id = arhgr.acl_res_id
+                        INNER JOIN `:db`.acl_res_group_relations argr
+                            ON argr.acl_res_id = res.acl_res_id
+                        INNER JOIN `:db`.acl_groups ag
+                            ON ag.acl_group_id = argr.acl_group_id
+                        WHERE ag.acl_group_id IN (:access_group_ids)
+                    )
+                    SQL;
+            }
+
+            if ($this->hasRestrictedAccessToHostCategories($accessGroupIds)) {
+                $hostCategoryAcls = <<<'SQL'
+                    AND hcr.hostcategories_hc_id IN (
+                        SELECT arhcr.hc_id
+                        FROM `:db`.acl_resources_hc_relations arhcr
+                        INNER JOIN `:db`.acl_resources res
+                            ON res.acl_res_id = arhcr.acl_res_id
+                        INNER JOIN `:db`.acl_res_group_relations argr
+                            ON argr.acl_res_id = res.acl_res_id
+                        INNER JOIN `:db`.acl_groups ag
+                            ON ag.acl_group_id = argr.acl_group_id
+                        WHERE ag.acl_group_id IN (:access_group_ids)
+                    )
+                    SQL;
+            }
+        }
+
+        $concatenatorA = new SqlConcatenator();
+        $concatenatorA->defineSelect(
```

Comment:
Same for ConcatenatorB, can you name the variable more explicitly ? So we could know while reading what is the real difference between both ?